src/types.rs

```diff
@@ -17,7 +18,12 @@ pub(crate) enum GossipMessage {
 	InitialSyncComplete,
 }
 
+#[derive(Clone, Copy)]
```

Comment:
I guess that's also an approach. What's the benefit of implicitly copying it on every pass as opposed to wrapping it in an Arc?